bittensor/commands/profile.py

```diff
@@ -0,0 +1,528 @@
+import os
+import yaml
+
+# 3rd Party
+from rich.prompt import Prompt
+from rich.table import Table
+
+# Bittensor
+import bittensor
+
+# Local
+from ..defaults import defaults
+
+SAVED_ATTRIBUTES = {
+    "profile": ["name"],
+    "subtensor": ["network", "chain_endpoint"],
+    "wallet": ["name", "hotkey", "path"],
+    "netuid": True,
+}
+
+
+def handle_error(message, exception):
+    bittensor.__console__.print(f":cross_mark: [red]{message}[/red]:[bold white] {exception}")
+
+
+def log_info(message):
+    bittensor.__console__.print(f":white_check_mark: [bold green]{message}[/bold green]")
+
+
+def log_warning(message):
+    bittensor.__console__.print(f":warning: [yellow]{message}[/yellow]")
+
+
+def log_error(message):
+    bittensor.__console__.print(f":cross_mark: [red]{message}[/red]")
+
+
+def list_profiles(path):
+    try:
+        os.makedirs(path, exist_ok=True)
+    except Exception as e:
+        handle_error("Failed to list profiles", e)
+        return []
+
+    try:
+        profiles = os.listdir(path)
+        profiles = [profile for profile in profiles if profile.endswith(".yaml") or profile.endswith(".yml")]
+    except Exception as e:
+        handle_error("Failed to list profiles", e)
+        return []
+
+    if not profiles:
+        log_error(f"No profiles found in {path}")
+        return []
+
+    return profiles
+
+
+def ask_for_profile(profiles, action, include_none=False):
+    profiles_without_extension = [profile.replace('.yml', '').replace('.yaml', '') for profile in profiles]
+
+    # Add 'None' as an option if include_none is True
+    if include_none:
+        profiles_without_extension.insert(0, 'None')
+
+    # If we have no profiles, return None
+    if len(profiles_without_extension) == 0:
+        return None
+
+    default_choice = 'None' if include_none else profiles_without_extension[0]
+    profile_name = Prompt.ask(
+        f"Enter profile name to {action}",
+        choices=profiles_without_extension,
+        default=default_choice
+    )
+
+    if profile_name == 'None':
+        return None
+
+    return profile_name + ('.yml' if profile_name + '.yml' in profiles else '.yaml')
+
+
+def get_profile_file_path(path, profile_name):
+    """Return the profile file path with the correct extension."""
+    profiles = list_profiles(path)
+    profile_base_name = profile_name.replace('.yml', '').replace('.yaml', '')
+
+    for profile in profiles:
+        if profile.startswith(profile_base_name):
+            return os.path.join(path, profile)
+
+    handle_error(f"Profile {profile_name} not found in {path}", None)
+    return None
+
+
+def get_profile_path_from_config(cli, action, include_none_option=False):
+    """Extract profile path from config and handle user prompt if necessary."""
+    config = cli.config.copy()
+    path = os.path.expanduser(config.profile.path)
+
+    if not (config.is_set("profile.name") and config.profile.name != config.profile.active) and not config.no_prompt:
```

Comment:
I thought a lot about the is_set function and the only reason I could think of is that if the value in the config is explicit None it would overwrite it with the default value.

Reply:
We no longer use a check like this in the profile command. We use whatever values are set in the config.